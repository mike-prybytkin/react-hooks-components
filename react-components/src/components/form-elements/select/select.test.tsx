import React from 'react';
import { render, screen } from '@testing-library/react';
import Select from './select';
import mockText from 'mocks/text';

const setUp = () =>
  render(<Select selectRef={undefined} labelType={mockText.labelUserSalary} id={'test'} />);

describe('Select component', () => {
  it('should correctly render select', () => {
    setUp();
    expect(screen.getByRole('combobox')).toBeInTheDocument();
  });

  it('should correctly set default option', () => {
    setUp();
    expect(
      (screen.getByRole('option', { name: 'less than 100$' }) as HTMLSelectElement).value
    ).toBe('less than 100$');
  });

  it('should display the correct number of options', () => {
    setUp();
    expect(screen.getAllByRole('option').length).toBe(7);
  });

  it('should create snapshot', () => {
    const { asFragment } = setUp();
    expect(asFragment()).toMatchSnapshot();
  });
});
