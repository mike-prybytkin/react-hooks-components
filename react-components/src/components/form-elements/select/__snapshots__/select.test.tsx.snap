// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should create snapshot 1`] = `
<DocumentFragment>
  <div
    class="select"
  >
    <label
      class="select__label"
    >
      Your monthly income:
      <select
        class="select__select"
        id="test"
      >
        <option
          disabled=""
        >
          Please choose an option
        </option>
        <option
          value="less than 100$"
        >
          less than 100$
        </option>
        <option
          value="100-199$"
        >
          100-199$
        </option>
        <option
          value="200-299$"
        >
          200-299$
        </option>
        <option
          value="300-399$"
        >
          300-399$
        </option>
        <option
          value="400-499$"
        >
          400-499$
        </option>
        <option
          value="over 500$"
        >
          over 500$
        </option>
      </select>
    </label>
  </div>
</DocumentFragment>
`;
